python/src/cucumber_messages/_messages.py

```diff
@@ -0,0 +1,954 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+
+class AttachmentContentEncoding(Enum):
+    identity = "IDENTITY"
+    base64 = "BASE64"
+
+
+class HookType(Enum):
+    before_test_run = "BEFORE_TEST_RUN"
+    after_test_run = "AFTER_TEST_RUN"
+    before_test_case = "BEFORE_TEST_CASE"
+    after_test_case = "AFTER_TEST_CASE"
+    before_test_step = "BEFORE_TEST_STEP"
+    after_test_step = "AFTER_TEST_STEP"
+
+
+class PickleStepType(Enum):
+    unknown = "Unknown"
+    context = "Context"
+    action = "Action"
+    outcome = "Outcome"
+
+
+class SourceMediaType(Enum):
+    text_x_cucumber_gherkin_plain = "text/x.cucumber.gherkin+plain"
+    text_x_cucumber_gherkin_markdown = "text/x.cucumber.gherkin+markdown"
+
+
+class StepDefinitionPatternType(Enum):
+    cucumber_expression = "CUCUMBER_EXPRESSION"
+    regular_expression = "REGULAR_EXPRESSION"
+
+
+class StepKeywordType(Enum):
+    unknown = "Unknown"
+    context = "Context"
+    action = "Action"
+    outcome = "Outcome"
+    conjunction = "Conjunction"
+
+
+class TestStepResultStatus(Enum):
+    unknown = "UNKNOWN"
+    passed = "PASSED"
+    skipped = "SKIPPED"
+    pending = "PENDING"
+    undefined = "UNDEFINED"
+    ambiguous = "AMBIGUOUS"
+    failed = "FAILED"
+
+
+@dataclass
+class Attachment:
+    """
+    //// Attachments (parse errors, execution errors, screenshots, links...)
+
+    *
+     An attachment represents any kind of data associated with a line in a
+     [Source](#io.cucumber.messages.Source) file. It can be used for:
+
+     * Syntax errors during parse time
+     * Screenshots captured and attached during execution
+     * Logs captured and attached during execution
+
+     It is not to be used for runtime errors raised/thrown during execution. This
+     is captured in `TestResult`.
+    """
+    """
+    *
+     The body of the attachment. If `contentEncoding` is `IDENTITY`, the attachment
+     is simply the string. If it's `BASE64`, the string should be Base64 decoded to
+     obtain the attachment.
+    """
+    body: str
+    """
+    *
+     Whether to interpret `body` "as-is" (IDENTITY) or if it needs to be Base64-decoded (BASE64).
+
+     Content encoding is *not* determined by the media type, but rather by the type
+     of the object being attached:
+
+     - string: IDENTITY
+     - byte array: BASE64
+     - stream: BASE64
+    """
+    content_encoding: "AttachmentContentEncoding"
+    """
+    *
+     The media type of the data. This can be any valid
+     [IANA Media Type](https://www.iana.org/assignments/media-types/media-types.xhtml)
+     as well as Cucumber-specific media types such as `text/x.cucumber.gherkin+plain`
+     and `text/x.cucumber.stacktrace+plain`
+    """
+    media_type: str
+    """
+    *
+     Suggested file name of the attachment. (Provided by the user as an argument to `attach`)
+    """
+    file_name: Optional[str] = None
+    source: Optional["Source"] = None
```

Comment:
[nit] we can use `Source` instead of `"Source"`, since we use `from __future__ import annotations`

Reply:
Done